pyiron_contrib/atomistic/atomicrex/atomicrex_job.py

```diff
@@ -44,23 +44,21 @@ def __init__(self, project, job_name):
         self._working_directory = None
         self.output = Output()
         self.factories = Factories()
-        
+
 
     def to_hdf(self, hdf=None, group_name=None):
-        if hdf == None:
-            hdf = self.project_hdf5
         super().to_hdf(hdf=hdf, group_name=group_name)
-        self.input.to_hdf(hdf=hdf)
-        self.potential.to_hdf(hdf=hdf)
-        self.structures.to_hdf(hdf=hdf)
+        self.input.to_hdf(hdf=self._hdf5)
```

Comment:
This also needs to pass the `group_name`.  When I look at the resulting HDF5 files it seems that the type information gets overwritten by subsequent calls from members.

Reply:
The `group_name` of the job type is already handled by the `super()`, so we could add the group_names of the specific classes, but that would be redundant.

Reply:
We might have to specify explicit groups the members to write themselves into, but that should be taken care of by the table_name of the input lists.  I'm looking at this right now.

Reply:
Exactly, if the `table_name` of the input list is set, then it is not necessary to define it again in the `to_hdf()` function, so the code is cleaner this way.